src/modules/iotjs_module_buffer.c

```diff
@@ -533,62 +564,71 @@ static jerry_value_t to_hex_string(const uint8_t* data, size_t length) {
   return ret_value;
 }
 
-static char to_base64_char(uint8_t digit) {
-  if (digit <= 25) {
-    return (char)digit + 'A';
-  }
-  if (digit <= 51) {
-    return (char)digit + 'a' - 26;
-  }
-  if (digit <= 61) {
-    return (char)digit + '0' - 52;
-  }
 
-  return (digit == 62) ? '+' : '/';
+static jerry_value_t to_base64_string(const uint8_t* data, size_t length) {
+  unsigned char* buffer = NULL;
+  size_t buffer_length = iotjs_base64_encode(&buffer, data, length);
+  jerry_value_t ret_value = jerry_create_string_sz(buffer, buffer_length);
+  IOTJS_RELEASE(buffer);
+
+  return ret_value;
 }
 
-static jerry_value_t to_base64_string(const uint8_t* data, size_t length) {
-  if (length == 0) {
-    return jerry_create_string_sz(NULL, 0);
-  }
+static const unsigned char base64_enc_map[65] =
+    "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/";
```

Comment:
Is this a perf or code size optimization?

Reply:
Perf. optimization